src/components/CartTable/index.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useMemo } from 'react'
+import { useDispatch } from 'react-redux'
+import { CartProduct, removeItemFromCart } from '../../store/cartSlice'
+import { Button, Table } from '../../styles/global'
+import AmmountInput from './AmmountInput'
+
+interface CartTableProps {
+  products: Array<CartProduct>
+}
+
+export default function CartTable ({ products }: CartTableProps) {
+  const dispatch = useDispatch()
+
+  const calculateTotal = useMemo(() => {
+    const total = products.reduce((prevValue, product) => prevValue + (product.ammount * product.price), 0)
+    return total.toFixed(2)
+  }, [products])
+
+  const removeItem = (product: CartProduct) => {
+    dispatch(removeItemFromCart({
```

Comment:
If you are not adding more properties to the product you don't need the spread operator.
```tsx
 dispatch(removeItemFromCart(product))
```